assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js

```diff
@@ -130,137 +140,145 @@ function SizePositionPanel({
+        <Area area="w">
+          <NumericInput
+            suffix={_x('W', 'The Width dimension', 'web-stories')}
+            value={width}
+            min={MIN_MAX.WIDTH.MIN}
+            max={MIN_MAX.WIDTH.MAX}
+            onChange={(evt) => {
+              const newWidth = Number(evt.target.value);
+              let newHeight = height;
+              if (lockAspectRatio) {
+                if (newWidth === '') {
+                  newHeight = '';
+                } else if (isNum(newWidth / origRatio)) {
+                  newHeight = dataPixels(newWidth / origRatio);
+                }
               }
-            }
-            pushUpdate(getUpdateObject(newWidth, newHeight));
-          }}
-          aria-label={__('Width', 'web-stories')}
-        />
-        <StyledToggle
-          aria-label={__('Aspect ratio lock', 'web-stories')}
-          title={__('Constrain proportions', 'web-stories')}
-          icon={<Locked />}
-          uncheckedIcon={<Unlocked />}
-          value={lockAspectRatio}
-          onChange={() => pushUpdate({ lockAspectRatio: !lockAspectRatio })}
-        />
-        <BoxedNumeric
-          suffix={_x('H', 'The Height dimension', 'web-stories')}
-          value={disableHeight ? '' : height}
-          placeholder={disableHeight ? __('AUTO', 'web-stories') : ''}
-          disabled={disableHeight}
-          min={MIN_MAX.HEIGHT.MIN}
-          max={MIN_MAX.HEIGHT.MAX}
-          onChange={(value) => {
-            const newHeight = value;
-            let newWidth = width;
-            if (lockAspectRatio) {
-              if (newHeight === '') {
-                newWidth = '';
-              } else if (isNum(newHeight * origRatio)) {
-                newWidth = dataPixels(newHeight * origRatio);
+              pushUpdate(getUpdateObject(newWidth, newHeight), true);
+            }}
+            aria-label={__('Width', 'web-stories')}
+            {...getMixedValueProps(width)}
+          />
+        </Area>
+        <Area area="d">
+          <Dash />
+        </Area>
+        <Area area="h">
+          <NumericInput
+            suffix={_x('H', 'The Height dimension', 'web-stories')}
+            value={disableHeight ? '' : height}
+            disabled={disableHeight}
+            min={MIN_MAX.HEIGHT.MIN}
+            max={MIN_MAX.HEIGHT.MAX}
+            onChange={(evt) => {
+              const newHeight = Number(evt.target.value);
+              let newWidth = width;
+              if (lockAspectRatio) {
+                if (newHeight === '') {
+                  newWidth = '';
+                } else if (isNum(newHeight * origRatio)) {
+                  newWidth = dataPixels(newHeight * origRatio);
+                }
               }
+              pushUpdate(getUpdateObject(newWidth, newHeight), true);
+            }}
+            aria-label={__('Height', 'web-stories')}
+            isIndeterminate={MULTIPLE_VALUE === height}
+            placeholder={
+              disableHeight
```

Comment:
For some reason I thought we had the `no-nested-ternary` rule in `eslint` on.
I'd still prefer splitting it up for readability but not a huge difference.

Reply:
Yeah, I wasn't a huge fan of this one.

Reply:
Fixed in ae6fbe42b.